### DISCUSSION TITLE
Getting Started with Seeed Studio XIAO SAMD21 | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#266] [Q&A]
# Getting Started with Seeed Studio XIAO SAMD21 | Seeed Studio Wiki

Getting Started with Seeed Studio XIAO SAMD21

https://wiki.seeedstudio.com/seeeduino-XIAO/#features

**Comment by gfmoore** [comment]
The Serial Plotting the DAC sketch has a SerialUSB command. I replaced this with Serial. It seemed to work. The output frequency is not 440Hz, more like 1.25Hz. The plotter works, but the output is not what you showed. It is mych slower and so you cannot see a full sine wave. I will try tweaking some values.

**Reply by gfmoore** [reply]
Ha, frequency isn't even used!

**Reply by JonPrevost** [reply]
Noticed the same thing today while doing a quick and dirty experiment seeing if this will replace an outdated Teensy 3.2 lol
The wisdom that, "you get what you pay for" applies in this case.
When I do an upload I can see the frequency going very fast on the scope, then it goes to this slower 2.73 Hz. Not sure what the problem is. I removed all the unnecessary code, just doing a DAC write. Glad I didn't spend much on this experiment.

**Reply by JonPrevost** [reply]
There are major glitches on this implementation. I scoped a simple analogWrite(A0, 0); analogWrite(A0, 1023); in the loop and major major missed pulses. Unacceptable. Looks like I either have another project on my hands or I need to just leave this alone for fewer grey hairs. Totally unimpressed with the DAC and examples by SEEED on this board. Shouldn't be selling it this way.

**Reply by MatthewJeffson** [reply]
Hi! Would this [PWM-DAC library](https://avdweb.nl/arduino/adc-dac/samd21-pwm-dac) library be helpful for your project? 
I'm sorry to reply so late.

**Reply by gfmoore** [reply]
It may, but I was just learning using the supplied sample :)

**Reply by WeSpeakEnglish** [reply]
Actual frequency about 1.37Hz. I have corrected the code (deleted not used variable and added a comment about the actual frequency in the new PR. Sorry for the inconvenience and the late update.
